java/code/src/com/redhat/rhn/frontend/xmlrpc/configchannel/ConfigChannelHandler.java

```diff
@@ -535,12 +535,12 @@ public ConfigRevision createOrUpdatePath(User loggedInUser,
         validKeys.add(ConfigRevisionSerializer.PERMISSIONS);
         validKeys.add(ConfigRevisionSerializer.REVISION);
         validKeys.add(ConfigRevisionSerializer.SELINUX_CTX);
+        validKeys.add(ConfigRevisionSerializer.CONTENTS_ENC64);
+        validKeys.add(ConfigRevisionSerializer.BINARY);
         if (!isDir) {
             validKeys.add(ConfigRevisionSerializer.CONTENTS);
-            validKeys.add(ConfigRevisionSerializer.CONTENTS_ENC64);
             validKeys.add(ConfigRevisionSerializer.MACRO_START);
             validKeys.add(ConfigRevisionSerializer.MACRO_END);
-            validKeys.add(ConfigRevisionSerializer.BINARY);
         }
```

Comment:
Thanks for the review @moio  ! The point is that _contents_enc64_ and _binary_ seems to be also valid for directories, indeed the error is:
`ERROR: redstone.xmlrpc.XmlRpcFault: Invalid argument(s): contents_enc64, binary`
caused by [https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/frontend/xmlrpc/BaseHandler.java#L433](https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/frontend/xmlrpc/BaseHandler.java#L433)  (missing validation of _contents_enc64_ and _binary_ keys althought they are present)

I think the reason why _contents_ is removed for directories is because it is always empty (instead it has a value for files) and probably it would cause problems in the further steps. 

I guess in the past when contents was empty _contents_enc64_ and _binary_ were not present, but it does not happen in my system (I am having my test on a SLES-15-SP3 installed using sumaform).

What do you think?

Reply:
I think @moio is right and the problem is not really in Java. Both options were added later but not cleanly removed when sending the info back to the server.
Maybe we better change the python code and remove also these two options if it is a directory.

Reply:
ok, got it! Thanks for your comments!

Reply:
Moved all the changes in the python part, mark the discussion as resolved. Thanks